File: server-java-play-framework/app/apimodels/CsvImportResponse.java
package apimodels;

import apimodels.NameStringValueResponse;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import com.fasterxml.jackson.annotation.*;
import java.util.Set;
import javax.validation.*;
import java.util.Objects;
import javax.validation.constraints.*;
import javax.validation.Valid;
/**
 * CsvImportResponse
 */
@javax.annotation.Generated(value = "org.openapitools.codegen.languages.JavaPlayFrameworkCodegen", date = "2026-02-09T20:49:20.380387445Z[Etc/UTC]", comments = "Generator version: 7.20.0-SNAPSHOT")
@SuppressWarnings({"UnusedReturnValue", "WeakerAccess"})
public class CsvImportResponse   {
  @JsonProperty("valid")
  
  private Boolean valid;

  @JsonProperty("message")
  
  private String message;

  @JsonProperty("version")
  
  private Double version;

  @JsonProperty("serializeNulls")
  
  private Boolean serializeNulls;

  @JsonProperty("startTimeLog")
  
  private Long startTimeLog;

  @JsonProperty("errorCode")
  
  private String errorCode;

  @JsonProperty("request")
  @Valid

  private List<@Valid NameStringValueResponse> request = null;

  @JsonProperty("batchId")
  
  private Long batchId;

  /**
   * Gets or Sets csvImportBatchStatus
   */
  public enum CsvImportBatchStatusEnum {
    NEW("NEW"),
    
    ERROR("ERROR"),
    
    COMPLETE("COMPLETE"),
    
    PARSE_ERROR("PARSE_ERROR"),
    
    PROCESSING("PROCESSING"),
    
    DUPLICATE("DUPLICATE");

    private final String value;

    CsvImportBatchStatusEnum(String value) {
      this.value = value;
    }

    @Override
    @JsonValue
    public String toString() {
      return String.valueOf(value);
    }

    @JsonCreator
    public static CsvImportBatchStatusEnum fromValue(String value) {
      for (CsvImportBatchStatusEnum b : CsvImportBatchStatusEnum.values()) {
        if (b.value.equals(value)) {
          return b;
        }
      }
      throw new IllegalArgumentException("Unexpected value '" + value + "'");
    }
  }

  @JsonProperty("csvImportBatchStatus")
  
  private CsvImportBatchStatusEnum csvImportBatchStatus;

  @JsonProperty("errorMessage")
  
  private String errorMessage;

  /**
   * Gets or Sets objectType
   */
  public enum ObjectTypeEnum {
    OFFERS("OFFERS"),
    
    RETAILERS("RETAILERS"),
    
    RETAILERLOCATIONS("RETAILERLOCATIONS"),
    
    CATEGORIES("CATEGORIES"),
    
    FILTERS("FILTERS");

    private final String value;

    ObjectTypeEnum(String value) {
      this.value = value;
    }

    @Override
    @JsonValue
    public String toString() {
      return String.valueOf(value);
    }

    @JsonCreator
    public static ObjectTypeEnum fromValue(String value) {
      for (ObjectTypeEnum b : ObjectTypeEnum.values()) {
        if (b.value.equals(value)) {
          return b;
        }
      }
      throw new IllegalArgumentException("Unexpected value '" + value + "'");
    }
  }

  @JsonProperty("objectType")
  
  private ObjectTypeEnum objectType;

  @JsonProperty("returning")
  
  private String returning;

  public CsvImportResponse valid(Boolean valid) {
    this.valid = valid;
    return this;
  }

   /**
   * Get valid
   * @return valid
  **/
  public Boolean getValid() {
    return valid;
  }

  public void setValid(Boolean valid) {
    this.valid = valid;
  }

  public CsvImportResponse message(String message) {
    this.message = message;
    return this;
  }

   /**
   * Get message
   * @return message
  **/
  public String getMessage() {
    return message;
  }

  public void setMessage(String message) {
    this.message = message;
  }

  public CsvImportResponse version(Double version) {
    this.version = version;
    return this;
  }

   /**
   * Get version
   * @return version
  **/
  public Double getVersion() {
    return version;
  }

  public void setVersion(Double version) {
    this.version = version;
  }

  public CsvImportResponse serializeNulls(Boolean serializeNulls) {
    this.serializeNulls = serializeNulls;
    return this;
  }

   /**
   * Get serializeNulls
   * @return serializeNulls
  **/
  public Boolean getSerializeNulls() {
    return serializeNulls;
  }

  public void setSerializeNulls(Boolean serializeNulls) {
    this.serializeNulls = serializeNulls;
  }

  public CsvImportResponse startTimeLog(Long startTimeLog) {
    this.startTimeLog = startTimeLog;
    return this;
  }

   /**
   * Get startTimeLog
   * @return startTimeLog
  **/
  public Long getStartTimeLog() {
    return startTimeLog;
  }

  public void setStartTimeLog(Long startTimeLog) {
    this.startTimeLog = startTimeLog;
  }

  public CsvImportResponse errorCode(String errorCode) {
    this.errorCode = errorCode;
    return this;
  }

   /**
   * Get errorCode
   * @return errorCode
  **/
  public String getErrorCode() {
    return errorCode;
  }

  public void setErrorCode(String errorCode) {
    this.errorCode = errorCode;
  }

  public CsvImportResponse request(List<@Valid NameStringValueResponse> request) {
    this.request = request;
    return this;
  }

  public CsvImportResponse addRequestItem(NameStringValueResponse requestItem) {
    if (this.request == null) {
      this.request = new ArrayList<>();
    }
    this.request.add(requestItem);
    return this;
  }

   /**
   * Get request
   * @return request
  **/
  public List<@Valid NameStringValueResponse> getRequest() {
    return request;
  }

  public void setRequest(List<@Valid NameStringValueResponse> request) {
    this.request = request;
  }

  public CsvImportResponse batchId(Long batchId) {
    this.batchId = batchId;
    return this;
  }

   /**
   * Get batchId
   * @return batchId
  **/
  public Long getBatchId() {
    return batchId;
  }

  public void setBatchId(Long batchId) {
    this.batchId = batchId;
  }

  public CsvImportResponse csvImportBatchStatus(CsvImportBatchStatusEnum csvImportBatchStatus) {
    this.csvImportBatchStatus = csvImportBatchStatus;
    return this;
  }

   /**
   * Get csvImportBatchStatus
   * @return csvImportBatchStatus
  **/
  public CsvImportBatchStatusEnum getCsvImportBatchStatus() {
    return csvImportBatchStatus;
  }

  public void setCsvImportBatchStatus(CsvImportBatchStatusEnum csvImportBatchStatus) {
    this.csvImportBatchStatus = csvImportBatchStatus;
  }

  public CsvImportResponse errorMessage(String errorMessage) {
    this.errorMessage = errorMessage;
    return this;
  }

   /**
   * Get errorMessage
   * @return errorMessage
  **/
  public String getErrorMessage() {
    return errorMessage;
  }

  public void setErrorMessage(String errorMessage) {
    this.errorMessage = errorMessage;
  }

  public CsvImportResponse objectType(ObjectTypeEnum objectType) {
    this.objectType = objectType;
    return this;
  }

   /**
   * Get objectType
   * @return objectType
  **/
  public ObjectTypeEnum getObjectType() {
    return objectType;
  }

  public void setObjectType(ObjectTypeEnum objectType) {
    this.objectType = objectType;
  }

  public CsvImportResponse returning(String returning) {
    this.returning = returning;
    return this;
  }

   /**
   * Get returning
   * @return returning
  **/
  public String getReturning() {
    return returning;
  }

  public void setReturning(String returning) {
    this.returning = returning;
  }


  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    CsvImportResponse csvImportResponse = (CsvImportResponse) o;
    return Objects.equals(valid, csvImportResponse.valid) &&
        Objects.equals(message, csvImportResponse.message) &&
        Objects.equals(version, csvImportResponse.version) &&
        Objects.equals(serializeNulls, csvImportResponse.serializeNulls) &&
        Objects.equals(startTimeLog, csvImportResponse.startTimeLog) &&
        Objects.equals(errorCode, csvImportResponse.errorCode) &&
        Objects.equals(request, csvImportResponse.request) &&
        Objects.equals(batchId, csvImportResponse.batchId) &&
        Objects.equals(csvImportBatchStatus, csvImportResponse.csvImportBatchStatus) &&
        Objects.equals(errorMessage, csvImportResponse.errorMessage) &&
        Objects.equals(objectType, csvImportResponse.objectType) &&
        Objects.equals(returning, csvImportResponse.returning);
  }

  @Override
  public int hashCode() {
    return Objects.hash(valid, message, version, serializeNulls, startTimeLog, errorCode, request, batchId, csvImportBatchStatus, errorMessage, objectType, returning);
  }

  @SuppressWarnings("StringBufferReplaceableByString")
  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class CsvImportResponse {\n");
    
    sb.append("    valid: ").append(toIndentedString(valid)).append("\n");
    sb.append("    message: ").append(toIndentedString(message)).append("\n");
    sb.append("    version: ").append(toIndentedString(version)).append("\n");
    sb.append("    serializeNulls: ").append(toIndentedString(serializeNulls)).append("\n");
    sb.append("    startTimeLog: ").append(toIndentedString(startTimeLog)).append("\n");
    sb.append("    errorCode: ").append(toIndentedString(errorCode)).append("\n");
    sb.append("    request: ").append(toIndentedString(request)).append("\n");
    sb.append("    batchId: ").append(toIndentedString(batchId)).append("\n");
    sb.append("    csvImportBatchStatus: ").append(toIndentedString(csvImportBatchStatus)).append("\n");
    sb.append("    errorMessage: ").append(toIndentedString(errorMessage)).append("\n");
    sb.append("    objectType: ").append(toIndentedString(objectType)).append("\n");
    sb.append("    returning: ").append(toIndentedString(returning)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}

File: server-kotlin-spring/src/main/kotlin/org/openapitools/model/CsvImportResponse.kt
package org.openapitools.model

import java.util.Objects
import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.annotation.JsonValue
import org.openapitools.model.NameStringValueResponse
import javax.validation.constraints.DecimalMax
import javax.validation.constraints.DecimalMin
import javax.validation.constraints.Email
import javax.validation.constraints.Max
import javax.validation.constraints.Min
import javax.validation.constraints.NotNull
import javax.validation.constraints.Pattern
import javax.validation.constraints.Size
import javax.validation.Valid
import io.swagger.v3.oas.annotations.media.Schema

/**
 * 
 * @param valid 
 * @param message 
 * @param version 
 * @param serializeNulls 
 * @param startTimeLog 
 * @param errorCode 
 * @param request 
 * @param batchId 
 * @param csvImportBatchStatus 
 * @param errorMessage 
 * @param objectType 
 * @param returning 
 */
data class CsvImportResponse(

    @Schema(example = "null", description = "")
    @get:JsonProperty("valid") val valid: kotlin.Boolean? = null,

    @Schema(example = "null", description = "")
    @get:JsonProperty("message") val message: kotlin.String? = null,

    @Schema(example = "null", description = "")
    @get:JsonProperty("version") val version: kotlin.Double? = null,

    @Schema(example = "null", description = "")
    @get:JsonProperty("serializeNulls") val serializeNulls: kotlin.Boolean? = null,

    @Schema(example = "null", description = "")
    @get:JsonProperty("startTimeLog") val startTimeLog: kotlin.Long? = null,

    @Schema(example = "null", description = "")
    @get:JsonProperty("errorCode") val errorCode: kotlin.String? = null,

    @field:Valid
    @Schema(example = "null", description = "")
    @get:JsonProperty("request") val request: kotlin.collections.List<NameStringValueResponse>? = null,

    @Schema(example = "null", description = "")
    @get:JsonProperty("batchId") val batchId: kotlin.Long? = null,

    @Schema(example = "null", description = "")
    @get:JsonProperty("csvImportBatchStatus") val csvImportBatchStatus: CsvImportResponse.CsvImportBatchStatus? = null,

    @Schema(example = "null", description = "")
    @get:JsonProperty("errorMessage") val errorMessage: kotlin.String? = null,

    @Schema(example = "null", description = "")
    @get:JsonProperty("objectType") val objectType: CsvImportResponse.ObjectType? = null,

    @Schema(example = "null", description = "")
    @get:JsonProperty("returning") val returning: kotlin.String? = null
) {

    /**
    * 
    * Values: NEW,ERROR,COMPLETE,PARSE_ERROR,PROCESSING,DUPLICATE
    */
    enum class CsvImportBatchStatus(@get:JsonValue val value: kotlin.String) {

        NEW("NEW"),
        ERROR("ERROR"),
        COMPLETE("COMPLETE"),
        PARSE_ERROR("PARSE_ERROR"),
        PROCESSING("PROCESSING"),
        DUPLICATE("DUPLICATE");

        companion object {
            @JvmStatic
            @JsonCreator
            fun forValue(value: kotlin.String): CsvImportBatchStatus {
                return values().firstOrNull{it -> it.value == value}
                    ?: throw IllegalArgumentException("Unexpected value '$value' for enum 'CsvImportResponse'")
            }
        }
    }

    /**
    * 
    * Values: OFFERS,RETAILERS,RETAILERLOCATIONS,CATEGORIES,FILTERS
    */
    enum class ObjectType(@get:JsonValue val value: kotlin.String) {

        OFFERS("OFFERS"),
        RETAILERS("RETAILERS"),
        RETAILERLOCATIONS("RETAILERLOCATIONS"),
        CATEGORIES("CATEGORIES"),
        FILTERS("FILTERS");

        companion object {
            @JvmStatic
            @JsonCreator
            fun forValue(value: kotlin.String): ObjectType {
                return values().firstOrNull{it -> it.value == value}
                    ?: throw IllegalArgumentException("Unexpected value '$value' for enum 'CsvImportResponse'")
            }
        }
    }

}

